lib/src/sidenav/Sidenav.tsx

```diff
@@ -1,56 +1,140 @@
-import React from "react";
+import React, { forwardRef, ReactElement, Ref, useMemo, useState } from "react";
 import styled, { ThemeProvider } from "styled-components";
-import { spaces, responsiveSizes } from "../common/variables.js";
+import { responsiveSizes } from "../common/variables.js";
+import { useResponsiveSidenavVisibility } from "../layout/SidenavContext";
 
 import useTheme from "../useTheme";
 import { BackgroundColorProvider } from "../BackgroundColorContext";
-import SidenavPropsType, { SidenavTitlePropsType, SidenavSubtitlePropsType, SidenavLinkPropsType } from "./types";
-import { useResponsiveSidenavVisibility } from "../layout/SidenavContext";
+import SidenavPropsType, {
+  SidenavGroupPropsType,
+  SidenavLinkPropsType,
+  SidenavSectionPropsType,
+  SidenavTitlePropsType,
+} from "./types.js";
+import DxcStack from "../stack/Stack";
 
-const DxcSidenav = ({ padding, children }: SidenavPropsType): JSX.Element => {
-  const colorsTheme = useTheme();
+const collapsableIcon = (
+  <svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="currentColor">
+    <path d="M0 0h24v24H0z" fill="none" />
+    <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
+  </svg>
+);
 
+const collapsedIcon = (
+  <svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="currentColor">
+    <path d="M0 0h24v24H0z" fill="none" />
+    <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
+  </svg>
+);
+
+const externalLinkIcon = (
+  <svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24" fill="currentColor">
+    <path d="M0 0h24v24H0z" fill="none" />
+    <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z" />
+  </svg>
+);
+
+const DxcSidenav = ({ children, title }: SidenavPropsType): JSX.Element => {
+  const colorsTheme = useTheme();
   return (
     <ThemeProvider theme={colorsTheme.sidenav}>
-      <SidenavContainer padding={padding}>
-        <BackgroundColorProvider color={colorsTheme.sidenav.backgroundColor}>{children}</BackgroundColorProvider>
+      <SidenavContainer>
+        <BackgroundColorProvider color={colorsTheme.sidenav.backgroundColor}>
+          {title}
+          <DxcStack divider={true} gutter="small">
```

Comment:
Sidenav's divider is going for the whole width of the component. Alex's design wasn't like it at all:

![image](https://user-images.githubusercontent.com/44321109/176680588-6b2868df-3ea4-4eea-8567-a35c4bbc687a.png)

Our implementation:

![image](https://user-images.githubusercontent.com/44321109/176680968-58f1a514-58f0-4b2f-b9e4-ae1735a45cc8.png)

Reply:
Yes, I also talked with Alex regarding this issue because it also affects the with of each link and group title that is collapsable.

Reducing the section width won't allow us to take the whole width for the links and groups.

Reply:
That's because we are currently using our layout components. First, I suggest not using the DxcStack since it's not a final version and will change shortly. And in case it is 100% necessary, to make the content of a section overflow the width set by the divider you can use the component DxcBleed